cardano-cli/src/Cardano/CLI/Json/Friendly.hs

```diff
@@ -234,39 +241,46 @@ friendlyTxBodyImpl
                  , "withdrawals" .= friendlyWithdrawals txWithdrawals
-                 , "governance actions"
-                    .= ( inEonForEra
-                          Null
-                          ( \(cOnwards :: ConwayEraOnwards era) ->
-                              case txProposalProcedures of
-                                Nothing -> Null
-                                Just (Featured _ TxProposalProceduresNone) -> Null
-                                Just (Featured _ (TxProposalProcedures lProposals _witnesses)) ->
-                                  friendlyLedgerProposals cOnwards $ toList lProposals
-                          )
-                          era
-                       )
-                 , "voters"
-                    .= ( inEonForEra
-                          Null
-                          ( \cOnwards ->
-                              case txVotingProcedures of
-                                Nothing -> Null
-                                Just (Featured _ TxVotingProceduresNone) -> Null
-                                Just (Featured _ (TxVotingProcedures votes _witnesses)) ->
-                                  friendlyVotingProcedures cOnwards votes
-                          )
-                          era
-                       )
-                 , "currentTreasuryValue" .= toJSON (unFeatured <$> txCurrentTreasuryValue)
-                 , "treasuryDonation" .= toJSON (unFeatured <$> txTreasuryDonation)
                  ]
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    ( \cOnwards ->
+                        case txProposalProcedures of
+                          Nothing -> []
+                          Just (Featured _ TxProposalProceduresNone) -> []
+                          Just (Featured _ (TxProposalProcedures lProposals _witnesses)) ->
+                            ["governance actions" .= (friendlyLedgerProposals cOnwards $ toList lProposals)]
+                    )
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    ( \cOnwards ->
+                        case txVotingProcedures of
+                          Nothing -> []
+                          Just (Featured _ TxVotingProceduresNone) -> []
+                          Just (Featured _ (TxVotingProcedures votes _witnesses)) ->
+                            ["voters" .= friendlyVotingProcedures cOnwards votes]
+                    )
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    (const ["currentTreasuryValue" .= toJSON (unFeatured <$> txCurrentTreasuryValue)])
+                    era
+                 )
+              ++ ( caseByronToBabbageOrConwaysEraOnwards
+                    (const ["treasuryDonation" .= toJSON (unFeatured <$> txTreasuryDonation)])
+                    era
```

Comment:
I think the code is easier to understand by not using `concat`, because it avoids nesting. And so it keeps the "one blob of code <==> one item in the list" structure that the items above have :+1: